Week one of UCS:JS

MattLuceen opened the pull request:
I opted for No Lecture because it will be interesting to see how much they remember without one. Muscle memory, hopefully?

I wish we had a way to do the code-along that only reveals the code as we write it, if that makes sense. If the students can't look ahead, we can ask them what we should type next... instead of them just reading it (or not reading it). 🤔

MattLuceen committed:
Add week one

MattLuceen committed:
Swap tabs for spaces for consistency

josephmaxwellhyland commented on BOOKREADME.md:
```diff
@@ -0,0 +1,27 @@
+- **5/18**: Final Project Presentations
+
+## Weekly Check-in
+[Click here to complete the weekly check-in survey.](https://forms.gle/7MzJBrf2W385tAvW6)
```
I think we will want to create a new form for this - or we can modify the existing form - or scrap it - I'm okay either way. For now, I think it's okay to just remove it from the page.

MattLuceen replied:
I made a new folder and a new weekly questionnaire in google forms (but nothing else, lol) but the link here is wrong :(

I will remove this for now and we can add these links in bulk later.

josephmaxwellhyland commented on BOOKREADME.md:
```diff
@@ -0,0 +1,27 @@
+[Click here to complete the weekly check-in survey.](https://forms.gle/7MzJBrf2W385tAvW6)
+
+## End-of-Trimester Survey/Assessment
+[Click here for the form.](https://forms.gle/NdJsPaDBwKruSb6VA)
```
Like above, we will want to update this form - but it can be removed for now :)

josephmaxwellhyland commented on HtmlCssReview/HtmlCssReviewCodeAlong.md:
````diff
@@ -0,0 +1,136 @@
+<h3>My Top Three Zones</h3>
+```
+
+## Add a List
````
I don't think we actually covered lists in the HTML/CSS trimester - we can probably just use paragraphs for this for simplicity

josephmaxwellhyland commented on HtmlCssReview/README.md:
```diff
@@ -0,0 +1,37 @@
+The [GPT-3 Mad Libs activity](https://platform.openai.com/playground/p/JQHsFLQt2yZuMc6VZfGDdR7K?model=text-davinci-003) is great for the first day because it allows the instructors to get familiar with saying and typing the students' names.
+
+1. Go around the room and have each student say their **name** and a **fact** about themself, and type that into the top half of the prompt (replacing the sample list)
+
```
small stylistic thing - i usually don't have newlines between list items

hylandtechoutreach commented:
> I wish we had a way to do the code-along that only reveals the code as we write it

![image](https://user-images.githubusercontent.com/43472480/216417936-5b280282-d514-4711-896d-517c484cac7d.png)

It's possible that I am making a mountain out of a mole hill -- maybe we just don't show them the page at all, and just have that for the instructor to read.

-- Matt

josephmaxwellhyland commented on HtmlCssReview/README.md:
```diff
@@ -0,0 +1,37 @@
+If you end up with time left over, have the students change the site to be about something else.
+
+## Kahoot
+Run through this [Kahoot](https://create.kahoot.it/share/9da0c22b-48ae-479f-9c7f-c0bfd458a2b5) to gauge what the students ultimately remember from HTML/CSS.
```
Kahoot was being slow for me so I couldn't check this quiz - but we will want to make sure it lines up with everything from the code-along

MattLuceen replied:
I meant to mention that the link wasn't working for me either. I stepped into developer tools and saw that the page was continuously reloading. I will investigate.

MattLuceen replied:
I think this is the last remaining item. I confirmed the issue exists on my iPhone too, so it's not some corporate web filtering nonsense breaking the site.

MattLuceen replied:
Resolved in the latest commit.

josephmaxwellhyland commented:
> I opted for No Lecture because it will be interesting to see how much they remember without one. Muscle memory, hopefully?

I think that works! the code-along is comprehensive, and they'll probably get back into the swing of things better by actually building something.

josephmaxwellhyland commented:
> > I wish we had a way to do the code-along that only reveals the code as we write it
> 
> ![image](https://user-images.githubusercontent.com/43472480/216417936-5b280282-d514-4711-896d-517c484cac7d.png)
> 
> It's possible that I am making a mountain out of a mole hill -- maybe we just don't show them the page at all, and just have that for the instructor to read.
> 
> -- Matt

yeah we don't have a great solution for this.

There's a hacky way to add a button that reveals code, like on this page: https://hylandtechclub.com/cs-101/MethodParameters/ParametersExercises.html

I will also hopefully be working on a way to dynamically show the instructions with a password - but that's the whole instruction page and not just the code itself.

If we had some front-end javascript loaded into each instruction page that was waiting for a message from a server somewhere, maybe we could do it dynamically... or something like that... but i think that would be a much larger undertaking and maybe an entirely new platform haha.

in general I think it's alright to either remove instructions entirely, or just let them see the code. If you want, you could also remove the code snippets from the instructions - that's up to you.

josephmaxwellhyland reviewed (commented):
Looks good to me! A couple minor things could be updated, but overall this should be good to go.

MattLuceen commented:
> There's a hacky way to add a button that reveals code, like on this page: https://hylandtechclub.com/cs-101/MethodParameters/ParametersExercises.html

I love it and am stealing it!

MattLuceen committed:
Fixup
- replaced links to forms with TODO
- added Click to Reveal Code buttons to code-along
- added styles for those buttons
- replaced list in code-along with another paragraph
- fixed newline consistency in readme

MattLuceen committed:
Fix Kahoot link
